test/check-types.js

```diff
@@ -106,6 +106,10 @@
             assert.isTrue(check.instance([], Array));
         });
 
+        test('instance with new Map and Map returns true', function() {
```

Comment:
Can you wrap this test inside an `if (typeof Map !== undefined)` check? It causes them to fail when run in older environments.